gobblin-service/src/test/java/org/apache/gobblin/service/modules/orchestration/OrchestratorTest.java

```diff
@@ -127,19 +133,31 @@ public void setup() throws Exception {
 
     SharedFlowMetricsSingleton sharedFlowMetricsSingleton = new SharedFlowMetricsSingleton(ConfigUtils.propertiesToConfig(orchestratorProperties));
 
-    this.orchestrator = new Orchestrator(ConfigUtils.propertiesToConfig(orchestratorProperties),
-        this.topologyCatalog, mockDagManager, Optional.of(logger), mockStatusGenerator,
-        Optional.of(mockFlowTriggerHandler), sharedFlowMetricsSingleton, Optional.of(mock(FlowCatalog.class)), Optional.of(dagManagementStateStore),
-        new FlowCompilationValidationHelper(config, sharedFlowMetricsSingleton, mock(UserQuotaManager.class), mockStatusGenerator));
-    this.topologyCatalog.addListener(orchestrator);
-    this.flowCatalog.addListener(orchestrator);
+    FlowCompilationValidationHelper flowCompilationValidationHelper = new FlowCompilationValidationHelper(config, sharedFlowMetricsSingleton, mock(UserQuotaManager.class), mockFlowStatusGenerator);
+    this.dagMgrNotFlowLaunchHandlerBasedOrchestrator = new Orchestrator(ConfigUtils.propertiesToConfig(orchestratorProperties),
```

Comment:
renamed to avoid confusion, as I now pass `Optional.absent()` for the `FlowLaunchHandler`.  nothing about the previously-existing tests below strictly validated that code path, so I made this entire `OrchestratorTest` class specific to the legacy `DagManager` version.

as for the lack of validation for the other code path, that suggests a missing test: e.g. that `orchestrate` invokes `FlowLaunchHandler::handleFlowLaunchTriggerEvent`

cc: @umustafi and @arjun4084346

Reply:
I can make a note to add the other code path for `FlowLaunchHandler baed orchestrate`. If those reside in this test class, then following the methodology discussed about we should have a separate `setupClass` method used for that test.